### ISSUE TITLE
Pow_Float does not conform to IEEE floating point standard (NaN^0 should be 1, not NaN)

### ISSUE THREAD

**Issue by Psychpsyo** [Yellow-Dog-Man/Resonite-Issues#1109]
### Describe the bug?

Using a node like Pow_Float to raise NaN to the power of 0 results in NaN.

### To Reproduce

Plug NaN and 0 into a Pow_Float node.

### Expected behavior

The output should be 1.

### Screenshots

![image](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/60073468/65865864-ccd6-47af-86d0-3f84e10e010c)

### Resonite Version Number

Beta 2024.1.3.1265

### What Platforms does this occur on?

Windows

### What headset if any do you use?

Vive, Desktop

### Log Files

I am not comfortable sharing my log files, especially for issues this easily replicated.

### Additional Context

The IEEE 754-2019 Standard for Floating-Point Arithmetic, section 9.2.1 says "**pow**(*x*, ±0) is 1 if *x* is not a signaling NaN".
(regular NaNs are not signaling NaNs)
Some programming languages (I've tried C, Python and Javascript) follow this convention.
C# does not and adds an explicit NaN check to its ``Math.Pow`` function.

### Reporters

cameron5 on Discord, Psychpsyo in Resonite

**Comment by Banane9** [comment]
Usually Resonite follows what C# does, so I'm not sure if there's gonna be a deviation from that behavior in this case 👀

**Comment by Frooxius** [comment]
Yeah we don't really have any special handling for stuff like this and we generally follow the C# API / spec. We wouldn't classify this as a bug, since this is deliberate change in the .NET Framework and according to the documentation, the specification allows disabling IEEE 754 exceptions, so it doesn't differentiate.

It looks like the behavior is changed on .NET x, so this will be changed implicitly once we do the switch: https://learn.microsoft.com/en-us/dotnet/api/system.math.pow?view=net-9.0

I'm going to close this one since we're really close to the switching that it's not worth making manual exception.
